### PR TITLE
[llvm] Update to 15.0.1

### LINKED ISSUE #26894: [llvm] update to 15
Library name: llvm

New version number: 15

Other information that may be useful (release notes, etc...)
https://github.com/llvm/llvm-project/releases/tag/llvmorg-15.0.1

### PR DESCRIPTION
**Pull request by joshchngs**
Updates the llvm port for LLVM 15.0.1 from 14.0.6

I recreated the existing patches so that the behaviour stays the same. However it looks like there either is now, or will be soon, a simpler way to achieve the same result, when @Ericson2314 has finished upstreaming the equivalent nixpkgs patches into LLVM mainline.
This work is part-merged in 15, so I figured it was best to keep the vcpkg patchset as-is until the upstream part stabilises - presumably in 16.

[nixpkgs patch status](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/llvm#information-on-our-current-patch-sets)
[LLVM merges from NixOs patches](https://reviews.llvm.org/differential/query/5UAfpj_9zHwY/)

- #### What does your PR fix?
  No issue.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Same as before, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by joshchngs** [comment]
I'm not sure what the bot is saying here. The version *is* updated vs master in `vcpkg.json`.

**Comment by JonLiu1993** [comment]
@joshchngs, Thanks for your pr, please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/78382/artifacts?artifactName=failure%20logs%20for%20x64-linux&api-version=7.0&%24format=zip). Please ping me if you need any help.

File: ports/llvm/0001-Fix-install-paths.patch [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ set(llvm_cmake_builddir "${LLVM_BINARY_DIR}/lib${LLVM_LIBDIR_SUFFIX}/cmake/llvm")
+diff --git a/lld/cmake/modules/CMakeLists.txt b/lld/cmake/modules/CMakeLists.txt
+index 57195bce45c9..e24c080725d7 100644
+--- a/lld/cmake/modules/CMakeLists.txt
++++ b/lld/cmake/modules/CMakeLists.txt
+@@ -5,13 +5,13 @@ include(FindPrefixFromConfig)
+ # Generate a list of CMake library targets so that other CMake projects can
+ # link against them. LLVM calls its version of this file LLVMExports.cmake, but
+ # the usual CMake convention seems to be ${Project}Targets.cmake.
+-set(LLD_INSTALL_PACKAGE_DIR "${CMAKE_INSTALL_PACKAGEDIR}/lld" CACHE STRING
```
Comment (abique) [inline ports/llvm/0001-Fix-install-paths.patch:86]: Why not just set `${CMAKE_INSTALL_PACKAGEDIR}` to `share`?

Comment (Ericson2314) [reply ports/llvm/0001-Fix-install-paths.patch:86]: That should work. We are using it in NixOS; other things need further patching but this part doesn't.

**Comment by JonLiu1993** [comment]
Pinging @joshchngs for response.

**Comment by joshchngs** [comment]
Sorry, I ended up just using Nix to bring the dependency in so I don't have a direct motivation to complete the work right now. If I have spare time in the future I may return to it, but I'm happy for somebody else to pick it up and apply the suggested changes if it's useful. If not feel free to close and I'll re-submit if needed.

**Comment by JonLiu1993** [comment]
Draft this PR since it seems that no progress is being made.

**Comment by JonLiu1993** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**